### PR TITLE
(feat)Analytics: Add support for individual experiment count

### PR DESCRIPTION
**Pull request by aditya109**
Signed-off-by: aditya109 <adikid1996@gmail.com>

**What this PR does / why we need it**:

- The bool-type **ANALYTICS** env is retrieved from `chaos-operator.yaml` and an enable-check is ran.
- On enabling, it sends GA request to out Mayadata `Litmus` instance updating the event regarding the individual experiment.

**Which issue this PR fixes**: [1042](https://github.com/litmuschaos/litmus/issues/1042)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -72,4 +72,4 @@ dockerops:
 	@echo "------------------"
 	@echo "--> Build Chaos-executor image..." 
 	@echo "------------------"
-	sudo docker build . -f build/Dockerfile -t litmuschaos/chaos-executor:ci
+	sudo docker build . -f build/Dockerfile -t daitya96/chaos-executor:test11
```
Comment (inyee786) [inline Makefile]: this is not needed

Comment (aditya109) [reply Makefile]: fixed this @inyee786

File: bin/exec.go [hunk 2]
```diff
@@ -216,6 +218,8 @@ func main() {
 		if err != nil {
 			log.Info("Unable to Update ChaosResult")
 			log.Error(err)
+		} else {
+			analytics.TriggerAnalytics(engineDetails.Experiments[i])
```
Comment (ksatchit) [inline bin/exec.go]: Why do we have it here ? We should be updating experiment counter irrespective of whether or not runner is able to update result. Would ideally do this before we start the experiment job?

Comment (aditya109) [reply bin/exec.go]: Changed the place of execution @ksatchit as requested

File: bin/exec.go [hunk 3]
```diff
@@ -4,12 +4,12 @@ import (
 	"flag"
 	"time"
 
+	"github.com/litmuschaos/chaos-executor/pkg/utils"
+	"github.com/litmuschaos/chaos-executor/pkg/utils/analytics"
+	"github.com/litmuschaos/chaos-operator/pkg/apis/litmuschaos/v1alpha1"
```
Comment (chandankumar4) [inline bin/exec.go]: Again you did the same mistake, It should be in the last where it was earlier

File: bin/exec.go [hunk 4]
```diff
@@ -167,7 +167,8 @@ func main() {
 		// Delete / retain the Job, using the jobCleanUpPolicy
 		err = utils.UpdateResultWithJobAndDeletingJob(engineDetails, resultName, experiment, clients)
 		if err != nil {
-			log.Infof("Unable to Update ChaosResult, error: %v", err)
-		}
+			log.Info("Unable to Update ChaosResult")
+			log.Error(err)
+		}
```
Comment (chandankumar4) [inline bin/exec.go]: Instead of these two lines use 
```
log.Errorf("Unable to Update ChaosResult %v", err)
```

File: pkg/utils/types.go [hunk 5]
```diff
@@ -20,6 +20,8 @@ type EngineDetails struct {
 	SvcAccount   string
 	AppKind      string
 	AppNamespace string
+	ClientUUID   string
+	Config       *rest.Config
```
Comment (chandankumar4) [inline pkg/utils/types.go]: Why we are adding `Config` in EngineDetails

File: build/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,5 @@
-FROM ubuntu:18.04
+# FROM ubuntu:18.04
+FROM registry.access.redhat.com/ubi7-dev-preview/ubi-minimal:7.6
```
Comment (chandankumar4) [inline build/Dockerfile]: Why this change is required?

File: bin/exec.go [hunk 7]
```diff
@@ -146,7 +146,7 @@ func main() {
 		experiment.JobName = engineDetails.Experiments[i] + "-" + randomString
 
 		log.Infof("JobName for this Experiment : %v", experiment.JobName)
-
+		analytics.TriggerAnalytics(engineDetails.Experiments[i])
```
Comment (ksatchit) [inline bin/exec.go]: place a comment here saying triggering  event before beginning job run

Comment (chandankumar4) [inline bin/exec.go]: Can you add check if clientUUID is non empty then it will run the analytics also move this in the starting of for loop with porper comments

Comment (aditya109) [reply bin/exec.go]: inserted the comment @ksatchit

File: pkg/utils/analytics/analytics.go [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+	// supported event categories
+
+	// category category notifies installation of a Litmus Experiment
+	category string = "Chaos-Experiment"
+
+	// supported event actions
+
+	// action is sent when the installation is triggered
+	action string = "Installation"
```
Comment (chandankumar4) [inline pkg/utils/analytics/analytics.go]: Change `Installation` to `Execution`

File: bin/exec.go (hunk 3 above)
Comment (aditya109) [reply bin/exec.go]: fixed it @chandankumar4 !
